src/udisksdaemonutil.c

```diff
@@ -73,6 +73,70 @@ void udisks_string_wipe_and_free (GString *string)
 }
 
 /**
+ * udisks_variant_lookup_binary:
+ * @dict: A dictionary #GVariant.
+ * @name: The name of the item to lookup.
+ * @out_text: (out): Return location for the binary text as #GString.
+ *
+ * Looks up binary data in a dictionary #GVariant and returns it as #GString.
+ *
+ * If the value is a bytestring ("ay"), it can contain arbitrary binary data
+ * including NUL values. If the value is a string ("s"), @out_text does not
+ * include the terminating NUL character.
+ *
+ * Returns: %TRUE if @dict contains an item @name of type "ay" that was
+ * successfully stored in @out_text, and %FALSE otherwise.
+ */
+gboolean
+udisks_variant_lookup_binary (GVariant     *dict,
+                              const gchar  *name,
+                              GString     **out_text)
+{
+  GVariant* item = g_variant_lookup_value (dict, name, NULL);
+  if (item)
+    return udisks_variant_get_binary (item, out_text);
+  return FALSE;
+}
+
+/**
+ * udisks_variant_get_binary:
+ * @value: A #GVariant of type "ay" or "s".
+ * @out_text: (out): Return location for the binary text as #GString.
+ *
+ * Gets binary data contained in a BYTEARRAY or STRING #GVariant and returns
+ * it as a #GString.
+ *
+ * If the value is a bytestring ("ay"), it can contain arbitrary binary data
+ * including NUL values. If the value is a string ("s"), @out_text does not
+ * include the terminating NUL character.
+ *
+ * Returns: %TRUE if @value is a bytestring #GVariant and was successfully
```

Comment:
Sorry for the inconsistency. These functions started off for bytestrings only before I realized it would be better to handle both types. Should I rename the functions as well?